TRAF4.py

```diff
@@ -0,0 +1,176 @@
+        return 100 * gc_sum / len(self.sequence)
+        
+    def __repr__(self):
+        return f'NucleicAcidSequence("{self.sequence}")'
+
+
+class DNASequence(NucleicAcidSequence):
+    def __init__(self, sequence):
+        super().__init__(sequence)
+        self.complement_dict = {
+            'a': 't', 'A': 'T',
+            't': 'a', 'T': 'A',
+            'g': 'c', 'G': 'C',
+            'c': 'g', 'C': 'G'
+        }
+        if 'U' in self.sequence.upper():
+            raise NuclAcidnucleotideError('U-contain sequence is not proper DNA sequence')
+
+    def transcribe(self):
+        transcription_dict = {
+        'a': 'a', 'A': 'A',
+        't': 'u', 'T': 'U',
+        'g': 'g', 'G': 'G',
+        'c': 'c', 'C': 'C'
+    }
+        result = RNASequence(''.join([transcription_dict[nuc] for nuc in self.sequence]))
+        return result
+
+    def __repr__(self):
+        return f'DNASequence("{self.sequence}")'
+
+
+class RNASequence(NucleicAcidSequence):
+    def __init__(self, sequence):
+        super().__init__(sequence)
+        self.complement_dict = {
+            'a': 'u', 'A': 'U',
+            'u': 'a', 'U': 'A',
+            'g': 'c', 'G': 'C',
+            'c': 'g', 'C': 'G'
+        }
+        if 'T' in self.sequence.upper():
+            raise NuclAcidnucleotideError('T-contain sequence is not proper RNA sequence')
+    
+    def __repr__(self):
+        return f'RNASequence("{self.sequence}")'
+
+
+class AminoAcidSequence(BiologicalSequence):
+    def gravy(self):
+        """Calculate GRAVY (grand average of hydropathy) value"""
+        gravy_aa_values = {'L': 3.8,
+                           'K': -3.9,
+                           'M': 1.9,
+                           'F': 2.8,
+                           'P': -1.6,
+                           'S': -0.8,
+                           'T': -0.7,
+                           'W': -0.9,
+                           'Y': -1.3,
+                           'V': 4.2,
+                           'A': 1.8,
+                           'R': -4.5,
+                           'N': -3.5,
+                           'D': -3.5,
+                           'C': 2.5,
+                           'Q': -3.5,
+                           'E': -3.5,
+                           'G': -0.4,
+                           'H': -3.2,
+                           'I': 4.5}
+        gravy_aa_sum = 0
+        for amino_ac in self.sequence.upper():
+            gravy_aa_sum += gravy_aa_values[amino_ac]
+        return round(gravy_aa_sum / len(self.sequence), 3)
+    
+    def __repr__(self):
+        return f'AminoAcidSequence("{self.sequence}")'
+
+
+
+def make_thresholds(threshold: int | float | tuple) -> tuple:
+    """Check thresholds input and convert single value to tuple"""
+    if isinstance(threshold, int) or isinstance(threshold, float):
+        lower = 0
+        upper = threshold
+    else:
+        lower = threshold[0]
+        upper = threshold[1]
+    return lower, upper
```

Comment:
Можно изменить `make_thresholds` для более явного понимания его назначения.

```suggestion
def make_thresholds(threshold: int | float | tuple) -> tuple:
    if isinstance(threshold, (int, float)):
        return 0, threshold
    elif isinstance(threshold, tuple):
        return threshold
    else:
        raise ValueError('Invalid threshold input')
```